pennylane/pauli/pauli_arithmetic.py

```diff
@@ -374,15 +378,28 @@ def operation(self, wire_order=None):
         return summands[0] if len(summands) == 1 else qml.sum(*summands)
 
     def hamiltonian(self, wire_order=None):
-        """Returns a native PennyLane :class:`~pennylane.Hamiltonian` representing the PauliSentence."""
+        """Returns a native PennyLane :class:`~pennylane.Hamiltonian` representing the PauliSentence.
+
+        .. warning::
+            Any complex components of the coefficients will be set to real as a PennyLane hamiltonian cannot have
+            complex coefficients!
+        """
         if len(self) == 0:
             if wire_order in (None, [], wires.Wires([])):
                 raise ValueError("Can't get the Hamiltonian for an empty PauliSentence.")
             return Hamiltonian([], [])
 
         wire_order = wire_order or self.wires
-        return sum(
-            coeff * pw.hamiltonian(wire_order=list(wire_order)) for pw, coeff in self.items()
+        wire_order = list(wire_order)
+
+        if any(npp.iscomplex(coeff) for coeff in self.values()):
+            raise ValueError(
+                "Can't get a Hamiltonian for a PauliSentence with non-zero complex coefficients"
+            )
+
+        return Hamiltonian(
+            npp.real(list(self.values())),
```

Comment:
Wouldn't a list of values be sufficient for Hamiltonian? We don't want to make things autograd when they aren't trainalble or different interfaces are being used in the circuit.

Reply:
Yes, but in reality we should expect them to be real. Also I was getting a weird error with gradients since the expectation value was complex in type, it was saying we need to use a holomorphic gradient.

Long story short: I removed this, I adjusted the failing test as it wasn't really trying to test complex coefficients in a hamiltonian anyways. Everything should work as expected. 🚀